internal/cmd/kafka/command_consumer_list.go

```diff
@@ -0,0 +1,72 @@
+package kafka
+
+import (
+	"github.com/spf13/cobra"
+
+	pcmd "github.com/confluentinc/cli/internal/pkg/cmd"
+	"github.com/confluentinc/cli/internal/pkg/errors"
+	"github.com/confluentinc/cli/internal/pkg/examples"
+	"github.com/confluentinc/cli/internal/pkg/kafkarest"
+	"github.com/confluentinc/cli/internal/pkg/output"
+)
+
+func (c *consumerCommand) newListCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List Kafka consumers.",
+		Args:  cobra.NoArgs,
+		RunE:  c.list,
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: `List all consumers for consumer-group "my-consumer-group".`,
+				Code: "confluent kafka consumer list --consumer-group my-consumer-group",
+			},
+		),
+	}
+
+	c.addConsumerGroupFlag(cmd)
+	pcmd.AddClusterFlag(cmd, c.AuthenticatedCLICommand)
+	pcmd.AddContextFlag(cmd, c.CLICommand)
+	pcmd.AddEnvironmentFlag(cmd, c.AuthenticatedCLICommand)
+	pcmd.AddOutputFlag(cmd)
+
+	cobra.CheckErr(cmd.MarkFlagRequired("consumer-group"))
+
+	return cmd
+}
+
+func (c *consumerCommand) list(cmd *cobra.Command, _ []string) error {
+	kafkaREST, err := c.GetKafkaREST()
+	if kafkaREST == nil {
+		if err != nil {
+			return err
+		}
+		return errors.New(errors.RestProxyNotAvailableMsg)
+	}
```

Comment:
I've been using this style instead for other commands. Is it really possible that both return values are `nil`, and if so, can we make `GetKafkaREST()` return the not available error in that case?
```suggestion
	if err != nil {
		return err
	}
```

Reply:
Looks like it can happen if a cluster's rest endpoint is missing. Is that possible?

Either way, I followed your suggestion and moved the error into the function. Link, mirror, and a few other Kafka commands still use this style and I'll update them in a followup PR after this one.

Reply:
Awesome, thank you! (Shouldn't be possible, but you never know...)